aea/helpers/install_dependency.py

```diff
@@ -20,54 +20,56 @@
 import subprocess  # nosec
 import sys
 from logging import Logger
+from subprocess import PIPE  # nosec
 from typing import List
 
 from aea.configurations.base import Dependency
 from aea.exceptions import AEAException, enforce
 
 
 def install_dependency(
-    dependency_name: str, dependency: Dependency, logger: Logger
+    dependency_name: str,
+    dependency: Dependency,
+    logger: Logger,
+    install_timeout: float = 300,
 ) -> None:
     """
     Install python dependency to the current python environment.
 
     :param dependency_name: name of the python package
     :param dependency: Dependency specification
     :param logger: the logger
+    :param install_timeout: timeout to wait pip to install
     """
     try:
-        pip_args = dependency.get_pip_install_args()
-        command = [sys.executable, "-m", "pip", "install", *pip_args]
-        logger.debug("Calling '{}'".format(" ".join(command)))
-        return_code = run_install_subprocess(command)
-        if return_code == 1:
-            # try a second time
-            return_code = run_install_subprocess(command)
-        enforce(return_code == 0, "Return code != 0.")
+        pip_args = dependency.get_call_pip_args()
+        logger.debug("Calling 'pip install {}'".format(" ".join(pip_args)))
+        call_pip(["install", *pip_args], timeout=install_timeout, retry=True)
     except Exception as e:
         raise AEAException(
             f"An error occurred while installing {dependency_name}, {dependency}: {e}"
         )
 
 
-def run_install_subprocess(
```

Comment:
Removing the method means we break backwards compatibility. Please keep method and then just call the new one?